README.md

```diff
@@ -87,6 +87,29 @@ docker image rm <image-id>
 
 There is no need to remove postgres and java or similar core images.
 
+## Cleanup
+
+Cleanup can be run in one of two modes for an environment:
+
+* ALWAYS_DELETE (default)
+* SKIP_RECENT_LOGINS
+
+ALWAYS_DELETE means that after the session expiry time any test data that has been created for that session will be deleted.
+
+SKIP_RECENT_LOGINS means that after the session expiry time any test data that has been created for that session will deleted,
+except for any users that have had a login within a certain time period. For any user that has recently logged in then their testing
+entity is set to DETACHED and their cleanup will no longer be managed by idam-testing-support-api.
+
+The configuration is controlled by the following attributes:
+
+* `cleanup.session.lifespan` (Duration) - After this duration a session can be selected for removal.
+* `cleanup.user.strategy` (ALWAYS_DELETE, SKIP_RECENT_LOGINS) - as above
+* `cleanup.user.recent-login-duration` (Duration) - When the cleanup strategy is set to SKIP_RECENT_LOGINS then this attribute is the window within
+which the user's last login time is considered "recent".
+
+Note that the `cleanup.user.recent-login-duration` cannot exceed the `cleanup.session.lifespan` and if it is set to a larger value then it will be overridden
```

Comment:
I'm not sure why we set this limitation?

Reply:
If the recent-login window is bigger than the session window you're looking for users who logged in before the testing session existed, which doesn't make any sense. If the session lifesspan is 2 weeks then we'd expect the users to be created at roughly the same time as the session, and then they're considered for cleanup at the end of the two weeks - for normal test users (that we probably want to clean up) you'd expect some activity on the first day, but then nothing after that. If you set the recent-login window to 1 week then you would exclude any users that had continued to be used after that initial activity.

Reply:
This user cleanup is a smaller part of the overall session cleanup - maybe that isn't clear?

Reply:
I think what I'm missing is that a user will actually be considered for deletion just once - when the testing session is over and never again.

Reply:
yes - then they're detached from the session